### PR TITLE
Allow configuration of non default port

### PR DESCRIPTION
**Pull request by brummer-simon**
This PR adds the possibility to set custom target port numbers deviating from 1883.

### CODE REVIEW

**Review by ryan-summers — CHANGES_REQUESTED** [review]
Hey, thanks for the PR! This looks wonderful - I hadn't considered using TLS with minimq since none of our network stacks support it, but indeed this library is agnostic to the underlying network stack and could well be used on networks that support TLS.

Can you please add an entry for this in the CHANGELOG.md file?

Other than that, I'd like to add one run-time check when trying to modify the broker port to prevent user foot-guns :)

File: src/mqtt_client.rs [hunk 1]
```diff
@@ -188,6 +187,17 @@ where
...
 
+    /// Set custom MQTT port to connect to.
+    ///
+    /// # Note
+    /// This must be completed before connecting to a broker.
+    ///
+    /// # Args
+    /// * `port` - The Port number to connect to.
+    pub fn set_broker_port(&mut self, port: u16) {
+        self.session_state.broker.set_port(port);
```
Comment (ryan-summers) [inline src/mqtt_client.rs:198]:
```suggestion
        if (*self.connection_state.state() != States::Restart) 
            return Err(Error::NotReady)'
        }
        
        self.session_state.broker.set_port(port);
```

**Comment by brummer-simon** [comment]
@ryan-summers: I'll add the changes. We don't have a proven Networkstack with TLS either, we try to piggy-back TLS ontop of a embedded_hal::TcpClientStack implementation. We will see if this works.

File: src/mqtt_client.rs (hunk 1 above)
Comment (ryan-summers) [reply src/mqtt_client.rs:198]: Correct - in the `Restart` state, we don't have any socket allocated yet from the stack. After we get it, we transition to `ConnectTransport` where we call the `TcpClientStack::connect()` command. If we just check for the `Restart` state, we can ensure that we haven't called `connect()` yet.

**Comment by brummer-simon** [comment]
@ryan-summers: All requested changes were added.

**Review by ryan-summers — APPROVED** [review]

**Pull request merged**